huaweicloud/services/acceptance/waf/resource_huaweicloud_waf_rule_blacklist_test.go

```diff
@@ -0,0 +1,177 @@
+package waf
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+
+	rules "github.com/huaweicloud/golangsdk/openstack/waf_hw/v1/whiteblackip_rules"
+)
+
+func TestAccWafRuleBlackList_basic(t *testing.T) {
+	var rule rules.WhiteBlackIP
+	randName := acctest.RandString(5)
+	rName1 := "huaweicloud_waf_rule_blacklist.rule_1"
+	rName2 := "huaweicloud_waf_rule_blacklist.rule_2"
+	rName3 := "huaweicloud_waf_rule_blacklist.rule_3"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.TestAccPreCheck(t) },
+		Providers:    acceptance.TestAccProviders,
+		CheckDestroy: testAccCheckWafRuleBlackListDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccWafRuleBlackList_basic(randName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckWafRuleBlackListExists(rName1, &rule),
+					resource.TestCheckResourceAttr(rName1, "ip_address", "192.168.0.0/24"),
+					resource.TestCheckResourceAttr(rName1, "action", "0"),
+
+					testAccCheckWafRuleBlackListExists(rName2, &rule),
+					resource.TestCheckResourceAttr(rName2, "ip_address", "192.165.0.0/24"),
+					resource.TestCheckResourceAttr(rName2, "action", "1"),
+				),
+			},
+			{
+				Config: testAccWafRuleBlackList_update(randName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckWafRuleBlackListExists(rName1, &rule),
+					resource.TestCheckResourceAttr(rName1, "ip_address", "192.168.0.125"),
+					resource.TestCheckResourceAttr(rName1, "action", "2"),
+
+					testAccCheckWafRuleBlackListExists(rName2, &rule),
+					resource.TestCheckResourceAttr(rName2, "ip_address", "192.150.0.0/24"),
+					resource.TestCheckResourceAttr(rName2, "action", "0"),
+				),
+			},
+			{
+				ResourceName:      rName3,
```

Comment:
suggest to use `rName1` as the ResourceName

Reply:
OK, it has been changed.